### ISSUE TITLE
HTML: don't remove http(s) from links

### ISSUE THREAD

**Issue by Knorkebrot** [tdewolff/minify#40]
I'm minifing HTML for a site that's available via HTTPS, but there's a link pointing to a site not supporting HTTPS. Minify removes "http:" from links like this:

```sh
> echo '<a href="http://kbct.de/">foo</a>' | minify -x .html
<a href=//kbct.de/>foo</a>
```

Now that link adopts the protocol from the page it's on, in my case HTTPS, which breaks the link.

**Comment by Knorkebrot** [comment]
Ah, I see, you want people to use the new rel="external" attribute on anchor elements. IMHO all links containing a domain are to be considered external, what do you think?

**Comment by tillberg** [comment]
I too had this issue. I serve my sites only over https, and I already specify protocol for links only when sites do not support https, so for my own use I just removed the block of code that deals with this: https://github.com/tillberg/minify/commit/cfaa3ee0697bd178a22d9bd47e7afb2760ad578a (I'd be happy to turn that into a PR if desired, but it doesn't seem friendly to just go and create a PR for something that basically just nukes someone else's code :) )

I agree with @Knorkebrot above. I get that there might be some use case for trying to strip the protocol off of pages that include them in self-references (e.g. some heavy server-side application frameworks do this, I think), but that seems like a small benefit to get compared to requiring everyone to mark up their anchors with rel="external" whenever they need to force the link protocol. (I also had never heard of rel="external" before, so I might be biased by my own ignorance.)

**Comment by tdewolff** [comment]
Yes you are right, it's not ideal. I will make changes to keep the protocol. It's a pity because stripping `http:` can considerably reduce the size.

**Comment by tdewolff** [comment]
Fixed in 3b07a59bb98162718bb9e53a83bd7a51964c971b on the develop branch. It will be re-enabled in the future when passing options is possible, including the currently used protocol. Either http or https can then be omitted, but not both.

**Comment by tdewolff** [comment]
Please check the `options` branch, it contains an implementation of passing options to the minifier. HTML now has the option `scheme` which can be either `http` or `https` and it will remove only that protocol for links. Call `m.Set("scheme", "http")` to enable that.
